### ISSUE TITLE
program dispatch data collection

### ISSUE THREAD

**Issue by pgkeller** [tenstorrent/tt-metal#10228]
We need a mechanism to collect and analyze data for program dispatch.  Ideally, we'd create histograms as well as view metrics for particular programs.  Data includes

1) Number of kernel groups
2) Number of kernel binary dispatcher transactions (dispatcher writes).  (this should cover memory spans per kernel bin)
3) Size of kernel binaries.  Would be good to view this in aggregate as well as by type (brisc/ncrisc/trisc*)
4) Number of runtime arg dispatcher writes
5) Total size of runtime args, min/max size of the transactions
6) Number of CB config dispatcher writes
7) Total size of CB configs, min/max size of the transactions
8) Number of semaphore dispatcher writes
9) Total size of semaphore writes, min/max size of the transactions

fyi @tt-aho 

Anything else?

**Comment by tt-dma** [comment]
PR https://github.com/tenstorrent/tt-metal/pull/11176 has a first-cut implementation for this. Next steps: 

- Get prefetcher data (reads + writes)
- Get data by op instead of by program to match device profiler (requires some outstanding changes on profiler side)
- Automate data colleciton

**Comment by pgkeller** [comment]
> PR #11176 has a first-cut implementation for this. Next steps:
> 
> * Get prefetcher data (reads + writes)
> * Get data by op instead of by program to match device profiler (requires some outstanding changes on profiler side)
> * Automate data colleciton

I don't think we need (yet) prefetcher read+write data.  This will generally line up w/ the API counts (eg, CBs, RTAs, kernel bins).  The most interesting thing about prefetcher data is if the code is behaving as we expect which we have other ways of determining and generally should be true.  The API data varies from model to model so will always be useful

**Comment by pgkeller** [comment]
this isn't done, but moving forward we'll go down a different path. closing, we'll open a new issue when the work is scheduled
